### ISSUE TITLE
default point lights seems wrong after exposure

### ISSUE THREAD

**Issue by mockersf** [bevyengine/bevy#11577]
## Bevy version

Main after https://github.com/bevyengine/bevy/pull/11347

## What went wrong

Looking at examples, most point lights are set to 150000 lumens, which is more than direct sunlight. Some are even up to 1000000 lumens, which seems an absurd value to set.

The default value of a point light is 800 lumens, which is now useless.

Bevy default values should make sense, and examples should not use values that don't make sense.

**Comment by GitGhillie** [comment]
Is the issue that the numbers are too big (i.e. you'd prefer a different scale/unit) or that they are not outputting the correct amount of light?

> most point lights are set to 150000 lumens, which is more than direct sunlight

Bit of a nitpick but that sounds like you're comparing lumens and lux which you can't do without a distance for reference. The sun outputs 36 octillion lumens...

**Comment by mockersf** [comment]
- try adding a point light, stick to the default
- see no change
- read the doc, it's lumens, check the defaults, 800 lumens, should be a 60W non-halogen incandescent bulb, sounds ok
- crank it up a bit, still no changes
- check the examples... 150000 lumens
- what kind of light bulb is that? where can I buy it??

We should use lux and lumens, and I don't know if the output is correct. My issue is that the defaults and the examples don't make sense

**Comment by GitGhillie** [comment]
Maybe good to note that in the 3d_scene for example we are trying to light the cube from almost 10m away, so that would require a powerful light. But yeah still not sure about the output either

**Comment by MAG-AdrianMeredith** [comment]
In layman's terms, lumens is the input energy, the others lux,candela etc is how much light reaches a particular surface.
If it helps I found the GLTF spec quite handy for under standing whats what for light intensities https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_lights_punctual/README.md

**Comment by tbillington** [comment]
Would it be worth having something similar to [`from_visibility`](https://docs.rs/bevy/0.12.1/bevy/pbr/enum.FogFalloff.html#method.from_visibility) for setting up the light ?

**Comment by cart** [comment]
Did some digging. Blender's default light is 1000 watts, their exporter exports this as 54,351 candelas (682,994 lumens). Blender's only exposure that I'm aware of / could find is the Color Management exposure value, which is calculated using `input_color * 2^exposure`. Note that this is the same formula Bevy uses for its Color Grading exposure. Both Blender and Bevy default these exposure values to 0.0.

This also means that #11347 added a _second_ exposure applied before the Color Grading exposure. This was not mentioned explicitly in that PR from what I've seen, and it does feel a bit strange to have two in two separate places. We should discuss / defend the existence of two exposure properties that exist in two different components that are functionally _very_ similar in terms of what they do to output colors.

I exported a scene from Blender to GLTF and tried to align everything I could:
* Blender defaults to Agx tonemapping, so I switched to that in Bevy
* Blender defaults to 0.0 exposure, so I set ColorGrading::exposure to 0.0 (this is our default as well)
* Blender, to my knowledge, has no other exposure settings. Therefore I hard coded `ExposureSettings::exposure()` to return 1.0, which makes it have no effect
* I removed ambient lighting on both sides (no light == pure black)
* Gamma is also 1.0 on both sides

Blender is on the left, Bevy is on the right:

![image](https://github.com/bevyengine/bevy/assets/2694663/234c284f-d237-40cf-b09a-48a9f514382b)

Bevy is significantly brighter / fully washed out. Something isn't being accounted for. Some ideas:

1. Maybe Blender actually has an additional hidden "camera exposure" that is darkening things (similar to our hard coded exposure prior to #11347). We would then just need to calibrate accordingly (aka we do need a second exposure as introduced in #11347). This _does_ make sense conceptually / is consistent with Filament _and_ our previous hard-coded exposure.
2. It is possible that the watts to candelas conversion done by blender is wrong (or their units are wrong somewhere else). Note that they _used to_ write watts directly to the gltf without conversion and then they corrected for that here: https://github.com/KhronosGroup/glTF-Blender-IO/issues/564. But I've seen a lot of conflicting conversations about converting Blender's watts to lumens (ex: [multiply by 683](https://blender.stackexchange.com/questions/45209/what-kind-of-units-does-the-cycles-emission-strength-use).
3. Maybe our lighting calculations are just very out of sync / don't agree on how a light of a particular power behaves.

Something that does feel a bit strange: if I _do_ write the watts value directly to the Bevy point light (1000 as the "intensity in lumens") with no other changes to the settings above, things line up pretty nicely:

![image](https://github.com/bevyengine/bevy/assets/2694663/2a4a98d6-c4d2-4713-aa7e-466916d20bb5)

(blender left, bevy right)

Lines up quite well in terms of brightness!

Likewise if I hard code Bevy's `ColorGrading::exposure` to `(-7.0f32).exp2() and set Blender's exposure to -7 (note the minus), we get this:

![image](https://github.com/bevyengine/bevy/assets/2694663/0f52b3d4-56f3-4371-8565-49d6169d6bfe)

They line up pretty much exactly! Maybe just coincidence. But it is interesting. If we plug our ears and pretend Watts == Lumens and that we don't need ev100 exposure settings, then are roughly in sync with blender _and_ we get lumen values that _line up with expectations from the real world_ (~1000 lumens for an indoor lamp). Weird.

Another comparison: if we take the first case (using the exported Watts->candelas->lumens), but also set our new `ExposureSettings::ev100` to 10.0 (more than the new default of 7.0, less than the old hard-coded default of ~12), we get this:

![image](https://github.com/bevyengine/bevy/assets/2694663/87aa04b6-d977-4cfc-8c0b-d76ab7020492)

Essentially identical!

And here is the current default ev100 (7.0):

![image](https://github.com/bevyengine/bevy/assets/2694663/78dade36-05eb-4fcd-8afe-47e942d339e3)

Note that Bevy's lighting is oversaturated relative to Blender's.

My current bet is:

1. The Blender GLTF exporter is probably correct about its watts -> candelas conversion
2. Eevee _does_ have some sort of internal exposure setting that is close to `ev100` 10.0 (which _is_ a very reasonable number)
3. For some reason inherent to the approach both we (and blender) are using, we need exorbitantly high / non-real-world lumens (on the order of hundreds of thousands) to emulate an indoor lamp (which should be closer to 1,000)

In that case, we should just calibrate our default `ev100` to 10.0, accept the crazy numbers required, and call it a day. It would be reassuring if we could reproduce these results in at least one other engine (blender, unity, unreal, etc) with aligned settings.

My "long shot" conspiracy theory bet (based on the weird agreement in the second test above) is that Eevee and Bevy are using roughly the same algorithms internally, they agree on lumens as units (and blender is just wrong about it being watts), blender doesn't have a hidden exposure roughly 10.0 ev100 (and therefore we should remove ours / default it to something that produces 1.0 as the factor), and we can therefore use "real world" lumen values.

**Comment by GitGhillie** [comment]
> This also means that https://github.com/bevyengine/bevy/pull/11347 added a second exposure applied before the Color Grading exposure. This was not mentioned explicitly in that PR from what I've seen, and it does feel a bit strange to have two in two separate places. We should discuss / defend the existence of two exposure properties that exist in two different components that are functionally very similar in terms of what they do to output colors.

There was a bit of discussion about that in the original PR: https://github.com/bevyengine/bevy/pull/8407#issuecomment-1871653253

**Comment by fintelia** [comment]
Please also see https://github.com/bevyengine/bevy/pull/8407#issuecomment-1858639312:

I think the reason both Blender and Bevy need seemingly huge values for light brightness is because CGI scenes use fewer lights and often place them further away from the objects they are lighting:
* Look around the space you're currently in. Is there only a single light bulb, or are there many lightbulbs? Remember that lumen is additive: 10x 800 lumen bulbs = 8000 lumens.
* Secondly, the inverse square falloff can be deceptively strong. The difference between a light 2 meters away and one 10 meters away is a factor of 25x!

There's also the lack of indirect light from point lights in scenes without walls (Blender) or any scene (Bevy), which I don't know how to quantify off-hand, but probably is a pretty significant impact

**Comment by cart** [comment]
Thanks for the links! I definitely missed the conversation that happened in #8407. Accidentally retread some ground :smile: 

I do agree that those points are relevant / play a role. But I find it _pretty_ hard to believe that the scene above (a 2 meter cube with a light 2 meters away) requires 682,994 lumens (341 times the power of my "lights up the whole room at night" 2000 lumen led bulb) to be reasonably lit from 2 meters away.

> There's also the lack of indirect light from point lights in scenes without walls (Blender) or any scene (Bevy), which I don't know how to quantify off-hand, but probably is a pretty significant impact

Hmmm yeah light bounces probably play a significant role here.

That being said, I think "agreement with Blender" is our highest priority. For 0.13 it probably makes sense to dial in the default ev100 to whatever gives us parity, which is essentially why @GitGhillie chose ev100 = 7 in 8407. Weirdly that value was very out of sync for me (see my above experiments). @GitGhillie, did you switch to AgX as your default tonemapper?

**Comment by cart** [comment]
I'll run some more tests

**Comment by GitGhillie** [comment]
Yeah I forgot about tonemapping back then but judging from the pictures I did remove the ambient light. Since I don't have the test setup from back then I made a new one (based on PR 11581 but should be the same) and interestingly I'm now also landing on an ev100 between 9 and 10 to match the lighting:
![image](https://github.com/bevyengine/bevy/assets/23659539/019c30c5-077a-4ebf-998d-46383980e0d3)

**Comment by cart** [comment]
Yup 10 seems like the sweet spot to me with the default 1000w light:

![image](https://github.com/bevyengine/bevy/assets/2694663/fdfb7e99-bcd0-423d-b069-50d2ecb5d78c)

Our reds desaturate way faster than Blender's for some reason. But everything else is pretty in sync.

**Comment by cart** [comment]
9.5 ev100 for reference (maybe slightly closer):
![image](https://github.com/bevyengine/bevy/assets/2694663/2b798627-19bd-4d33-9087-4670272c7b5c)

**Comment by cart** [comment]
10w light, 9.5 ev100 (we end up being "too bright" in some cases)
![image](https://github.com/bevyengine/bevy/assets/2694663/6199de2f-5123-46b4-b70a-53713a91698d)

**Comment by cart** [comment]
10w light, 10 ev100 (better, but green still shows "too much" relative to blender)
![image](https://github.com/bevyengine/bevy/assets/2694663/33c7ca53-d296-41b0-ae3d-cb5eccecdf1c)

**Comment by cart** [comment]
10w light, 10.5 ev100 (we start seriously losing the white cube and blue sphere relative to blender, green is still slightly too visible)
![image](https://github.com/bevyengine/bevy/assets/2694663/adf8a3b3-754c-4db8-bc8b-9c2e76d094f4)

My take is 10 ev100 is the best for point lights

**Comment by cart** [comment]
Directional light 1 W/m^2, ev100 10:
![image](https://github.com/bevyengine/bevy/assets/2694663/47ec5b9d-8e18-4e77-b48a-e81c87803554)

Blender is _slightly_ brighter across the board

**Comment by cart** [comment]
Directional light 1 W/m^2, ev100 9.5:
![image](https://github.com/bevyengine/bevy/assets/2694663/d32da7c8-1609-4a18-8560-d52b6e91d3c7)

Now we're slightly brighter than blender. 9.5 seems closer than 10, sweet spot at this brightness is probably 9.7 or something.

**Comment by cart** [comment]
Directional light 0.05 W/m^2, ev100 9.5

![image](https://github.com/bevyengine/bevy/assets/2694663/3859c6a9-60e2-4bad-93f2-3061a1138c47)

We are significantly darker. ev100 9.2 seems like the sweet spot:

![image](https://github.com/bevyengine/bevy/assets/2694663/5f3b3a20-8b07-44a4-9910-13c5a5bf7ef9)

**Comment by cart** [comment]
10 W/m^2, ev100 9.2

![image](https://github.com/bevyengine/bevy/assets/2694663/ba9e0e07-68e8-4713-a46c-ad68bd624cb8)

We are slightly brighter (and significantly more desaturated).

ev100 10.0

![image](https://github.com/bevyengine/bevy/assets/2694663/1ff51394-71d2-4ba6-b917-0ca1e3316847)

We are slightly darker.

ev100 9.5 feels close to the sweet spot
![image](https://github.com/bevyengine/bevy/assets/2694663/e41aa1dd-f141-431e-a0e2-5e9c4a621ecc)

**Comment by fintelia** [comment]
I'm going to guess that the real value is log2(1000 / 1.2) = 9.703.

**Comment by cart** [comment]
My takeaway is that there isn't a perfect value to achieve parity, but that the ideal constraint solve is _in fact_ approximately 9.7 :)

**Comment by cart** [comment]
For 0.13, I think we set the default to 9.7 (with a link to this issue in the docs for posterity), then adjust brightness of the lights in the examples accordingly with their end-of-series DragonballZ-tier power levels.

**Comment by GitGhillie** [comment]
@doonv how do you feel about changing the default in #11581 again? 😅 
Otherwise I can also make a PR to your PR tomorrow (I'm on EU time, need to go now)

**Comment by doonv** [comment]
> @doonv how do you feel about changing the default in #11581 again? 😅 Otherwise I can also make a PR to your PR tomorrow (I'm on EU time, need to go now)

I'd rather not, I already went through the examples multiple times and I don't wanna do that again.

**Comment by GitGhillie** [comment]
Understandable! I'm willing to do it but first I want to see a go/no-go decision on your PR before I also start going back and forth on the examples

**Comment by fintelia** [comment]
Blender seems to be assuming 1 W = 1 lumen and applying exposure by default (errors which cancel each other out if the true exposure is ev100=9.7). Perhaps the thing to do is to add a constant of `BLENDER_DEFAULT = 9.7` and not even pretend it is physically based?

**Comment by cart** [comment]
Given that the purpose of the constant is to sync with blender, I'm cool with that (although it seems premature to say if / how blender is getting things wrong). 9.7 seems like a pretty reasonable "middle of the road" exposure value.

**Comment by cart** [comment]
(but for the record, it certainly seems like we're missing _something_ here either on the Blender side or the Bevy side)

**Comment by cart** [comment]
Strategy-wise, lets finish reviewing and merging #11581 and then add `EV100_BLENDER` to the constants, use it as default, and in the same PR re-calibrate the examples

**Comment by fintelia** [comment]
Is that going to work with Blender exported GLTFs? Or does `EV100_BLENDER` actually need to be `9.7 + log2(683) = 19.118`?

**Comment by DGriffin91** [comment]
It's also worth noting that blender uses a different version of AgX than bevy does. Bevy added AgX before blender did. Bevy uses the [original AgX](https://github.com/sobotka/AgX) getting it's LUT in a easier format from https://github.com/MrLixm/AgXc. [Note on AgXc github about blender's version.](https://github.com/MrLixm/AgXc?tab=readme-ov-file#background)

Subjectively, they look a bit different, it's been a while since I compared so I don't remember what the differences were.

Blender uses https://github.com/EaryChow/AgX

Blender AgX release notes:
https://wiki.blender.org/wiki/Reference/Release_Notes/4.0/Color_Management

Blender PR that added AgX
https://projects.blender.org/blender/blender/pulls/106355

It would probably be good to get blender's version in bevy at some point. Idk about the licensing, AgX, AgXc and Eary's AgX all don't have any license documents of any kind.

If you turn off tonemapping or use blender filmic in both they should match a bit more closely.

**Comment by cart** [comment]
> Is that going to work with Blender exported GLTFs? Or does EV100_BLENDER actually need to be 9.7 + log2(683) = 19.118?

`EV100_BLENDER` would be 9.7 because thats what I set `ExposureSettings::ev100` to when I was calibrating.

**Comment by cart** [comment]
(and yes that calibration was done with Blender-exported gltfs)

**Comment by cart** [comment]
> It's also worth noting that blender uses a different version of AgX than bevy does. Bevy added AgX before blender did. Bevy uses the [original AgX](https://github.com/sobotka/AgX) getting it's LUT in a easier format from https://github.com/MrLixm/AgXc. [Note on AgXc github about blender's version.](https://github.com/MrLixm/AgXc?tab=readme-ov-file#background)

Ahhh gotcha that would explain it

**Comment by cart** [comment]
Blender Filmic, ev100 9.7, directional light 10 W/m^2
![image](https://github.com/bevyengine/bevy/assets/2694663/1921a2a8-f89f-4c2a-8d9c-37422e38e7a1)

(basically identical)

directional light 0.03 W/m^2

![image](https://github.com/bevyengine/bevy/assets/2694663/3af8df65-ede2-4e00-887c-2e0bd7c5e13c)

(basically identical)

point light 10w

![image](https://github.com/bevyengine/bevy/assets/2694663/72323cf3-ddc6-4767-863b-edd0582423d2)

(big divergence)

point light 500w

![image](https://github.com/bevyengine/bevy/assets/2694663/305f8a0f-80a3-47f5-9983-9aafd2d850dd)

(basically identical)

Consensus: basically identical with the exception of low-light point lights.

**Comment by cart** [comment]
![image](https://github.com/bevyengine/bevy/assets/2694663/11da1717-5670-4755-bf0c-cbef0e0508ff)

The big "DragonBallZ" units are actually probably correct.

**Comment by cart** [comment]
Just kicked off a merge of #11581. I'm doing the follow up work we discussed above now.

**Comment by cart** [comment]
Reopening this because it isn't resolved yet
